File: backend/database_stuff.py
from os import environ as ENV
from psycopg_pool import AsyncConnectionPool

connection_pool = AsyncConnectionPool(f"host={ENV['CHESSBOARDNET_DATABASE_HOST']} dbname=chessboardnet user=chessboardnet password={ENV['CHESSBOARDNET_DATABASE_PASSWORD']}",open=False)

class NoLogin(Exception):
	pass

async def on_startup(app):
	await connection_pool.open()

async def get_user(login_token):
	async with connection_pool.connection() as db:
		cur = await db.execute("select user_id, username from users where login_token=%s",[login_token])

		async for user_id, username in cur:
			return user_id, username

	raise NoLogin

File: backend/database_stuff.js
import pg from 'pg'

export const pool = new pg.Pool({
	user: process.env.CHESSBOARDNET_DATABASE_USER,
	password: process.env.CHESSBOARDNET_DATABASE_PASSWORD,
	host: process.env.CHESSBOARDNET_DATABASE_HOST,
	database: process.env.CHESSBOARDNET_DATABASE_NAME,
	port:5432,
})

export async function get_user(login_token) {
	var result = await pool.query("select * from users where login_token=$1 and login_expires>now()",[login_token])

	if(result.rowCount==1)
		return result.rows[0]
	else
		return null
}

